src/openrct2/object/ObjectFactory.cpp

```diff
@@ -440,6 +440,7 @@ namespace ObjectFactory
             result->ReadJson(&readContext, jRoot);
             if (readContext.WasError())
             {
+                delete result;
```

Comment:
This does fix this specific instance but we still need to track that this needs conversion to using something like a unique ptr. This PR is not to be merged without making an issue for that.

Reply:
Please see #13100